snapcraft/how-to/qt-applications/qt-applications.md

```diff
@@ -1,265 +1,486 @@
+    command: usr/bin/kcalc
     extensions:
       - kde-neon
     plugs:
-      - home
-      - opengl
-      - network
-      - network-bind
-      - pulseaudio
+      - audio-playback
 ```
 
-You can see we use the [`kde-neon` extension](/t/the-kde-neon-extension/13752). This extension will make Qt5 and KDE libraries available to the snap at run time and it will configure the run time environment of the application so that all desktop functionality is correctly initialised.
+(If you followed the 'Cursor themes' part of this guide, the resulting `apps` section will also include a `command-chain` entry.)
 
-The `common-id` field is used to link the AppStream metadata to this application. As a result, we don't need to [manually specify the `.desktop` entry file](/t/desktop-files-for-menu-integration/13115) because it's already defined in AppStream. See [Using AppStream metadata](/t/using-external-metadata/4642#heading--appstream) for more information.
+The `audio-playback` interface enables an app to play sounds whilst under `strict` confinement. It isn't necessary when testing a snap under `devmode` confinement, as `devmode` allows snaps to access any interface it needs, so if you're having sound-related issues in `devmode` it's likely that there is another issue (such as a missing library, which is also the case here).
 
+KCalc plays error sounds using KDE's *KNotification* Framework, which in turn relies upon the *libcanberra* audio system. The `kde-neon` extension provides the main *libcanberra* library, but not the required *libcanberra-pulse* plugin. 
 
-### Building the snap
+As the plugin is contained in the `libcanberra-pulse` package in the `core22` software archive, we *could* use `stage-packages` to download and install the package, in the same way we did for the MPFR and GMP libraries earlier on. Snapcraft would then fetch that package for us, as well as any other library or data packages that it depends on, at the start of the build.
 
-You can download the example repository with the following command:
+However, we don't need Snapcraft to download each of the dependencies of *libcanberra-pulse* as they're already contained in the Qt/KDE content snap. Instead, to keep the size of our snap as small as possible, we can use `override-pull` and `override-build` to tell Snapcraft how to download and extract just the additional package that we're interested in.
 
-```bash
-$ git clone https://github.com/galgalesh/kcalc.git
+We can do this by adding a new part to *snapcraft.yaml* named `sound-support` as follows:
+
+```yaml
+  sound-support:
+    plugin: nil
+    override-pull: |
+      if [ ! -e libcanberra-pulse_*.deb ]; then
+        apt-get download libcanberra-pulse:$CRAFT_ARCH_BUILD_FOR
+      fi
+    override-build: |
+      if [ ! -e $CRAFT_PART_INSTALL/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/libcanberra-*/libcanberra-pulse.so ]; then
+        dpkg -x $CRAFT_PART_SRC/libcanberra-pulse_*.deb $CRAFT_PART_INSTALL
+      fi
```

Comment:
This is also a bit of a hack, but it works...

I know that there's a drop-in `cleanup` part that removes duplicate files, but I liked the idea of avoiding the duplication in the first place.